components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/dao/impl/IdentityUserMetadataMgtDAOImpl.java

```diff
@@ -0,0 +1,508 @@
+                                                      String tenantId, String username, String claimAttributeName,
+                                                      String value) throws SQLException, IdentityEventException {
+
+        String sqlStmt = fetchSQLStatementFromUserStoreConfig(userStoreManager,
+                IdentityRecoveryConstants.SQLQueries.UPDATE_USER_METADATA_TO_USER_STORE_SQL_KEY);
+
+        PreparedStatement prepStmt = connection.prepareStatement(sqlStmt);
+        prepStmt.setString(1, value);
+        prepStmt.setString(2, username);
+        prepStmt.setString(3, tenantId);
+        prepStmt.setString(4, claimAttributeName);
+        prepStmt.setString(5, UserCoreConstants.DEFAULT_PROFILE);
+        prepStmt.setString(6, tenantId);
+        int result = prepStmt.executeUpdate();
+        return result > 0;
+    }
+
+    /**
+     * Insert new user metadata in user store.
+     *
+     * @param userStoreManager      UserStoreManager.
+     * @param tenantId              Tenant Id.
+     * @param username              Username.
+     * @param claimAttributeName    Claim attribute name.
+     * @param value                 Value of the claim.
+     * @return                      boolean value of the execution.
+     * @throws SQLException         SQL Exception.
+     */
+    private boolean insertMetadataInUserStore(Connection connection, UserStoreManager userStoreManager, String tenantId,
+                                              String username, String claimAttributeName, String value)
+            throws SQLException, IdentityEventException {
+
+        String sqlStmt = fetchSQLStatementFromUserStoreConfig(userStoreManager,
+                IdentityRecoveryConstants.SQLQueries.INSERT_USER_METADATA_TO_USER_STORE_SQL_KEY);
+
+        PreparedStatement prepStmt = connection.prepareStatement(sqlStmt);
+        prepStmt.setString(1, username);
+        prepStmt.setString(2, tenantId);
+        prepStmt.setString(3, claimAttributeName);
+        prepStmt.setString(4, value);
+        prepStmt.setString(5, UserCoreConstants.DEFAULT_PROFILE);
+        prepStmt.setString(6, tenantId);
+        int result = prepStmt.executeUpdate();
+        return result > 0;
+    }
+
+    /**
+     * Get database connection.
+     *
+     * @param realmConfiguration        Realm configuration.
+     * @return                          Database connection.
+     * @throws IdentityEventException   Identity event exception.
+     */
+    private Connection getDBConnection(RealmConfiguration realmConfiguration) throws IdentityEventException {
+
+        Connection dbConnection = null;
+        DataSource dataSource = DatabaseUtil.createUserStoreDataSource(realmConfiguration);
+
+        try {
+            if (dataSource != null) {
+                dbConnection = DatabaseUtil.getDBConnection(dataSource);
+            }
+
+            //if primary user store, DB connection can be same as realm data source.
```

Comment:
Formatting issue.

Reply:
Fixed in https://github.com/wso2-extensions/identity-governance/pull/697/commits/dd3419fd69cc40e847e40ed9dfb29c867a3f0018